Framework/API/inc/MantidAPI/ISpectrum.h

```diff
@@ -105,7 +105,7 @@ class DLLExport ISpectrum {
     HistogramData::Histogram histogram(std::forward<T>(data)...);
     // Check for the special case EventList, it only accepts histograms without
     // Y and E data.
-    checkHistogram(histogram);
+    checkAndSanitizeHistogram(histogram);
     mutableHistogramRef() = std::move(histogram);
```

Comment:
And we would not just want to silently drop Y and E data. Raising an exception is safer.